src/app/components/button/ButtonWithShadow.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+
+import PropTypes from "prop-types";
+
+const propTypes = {
+    buttonText: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+    isSubmitting: PropTypes.bool,
+    class: PropTypes.string,
+    disabled: PropTypes.bool,
+};
+
+const ButtonWithShadow = props => (
+    <button type={props.type} className={`ons-btn ${props.class ? "ons-btn--" + props.class : ""}`} onClick={props.onClick} disabled={props.disabled}>
+        <span className="ons-btn__inner">
+            {props.buttonText}
+            {props.isSubmitting && (
+                <svg
```

Comment:
any reason you can't use existing `ButtonWithSpinner` ? If not you need to extract svgs to separate file and reuse it. In form like that it is very messy
src/app/icons
again this comment was lest before